vj4/ui/templates/problem_detail.html

```diff
@@ -44,18 +44,34 @@ <h1 class="section__title">{{ _('Information') }}</h1>
       </div>
     </div>
     <div class="section side">
-      <div class="section__indent top bottom">
-        <h1 class="section__title">{{ _('Related') }}</h1>
-        <div class="section__indent left right">
-          <div class="typo">
-            <p>以下题单收录：</p>
-            <p>123abc</p>
-            <p>处于以下训练计划中：</p>
-            <p>123abc</p>
-          </div>
+      <h1 class="section__title">{{ _('Related') }}</h1>
+      <div class="typo">
+        <p>以下题单收录：</p>
+        <p>123abc</p>
+        <p>处于以下训练计划中：</p>
+        <p>123abc</p>
+      </div>
+    </div>
+  </div>
+</div>
+
+<!-- IDE Mode -->
+<div class="ide ide-container">
+  <div class="ide-layout--table">
+    <div class="ide-layout--table-row">
+      <!-- problem --><div class="ide-layout--table-cell section">
+        <div class="section__indent top bottom left right">
+          <h1>{{ pdoc['title'] }}</h1>
```

Comment:
you should at least copy the DOM when entering IDE mode, instead of outputing contents twice.